roofit/roofit/src/RooSDSCBShape.cxx

```diff
@@ -0,0 +1,246 @@
+ * @(#)root/roofit:$Id$
+ * Authors:                                                                  *
+ *    T. Skwarnicki modify RooCBShape to Symmetrical Double-Sided CB         *
+ *    Michael Wilkinson add to RooFit source                                 *
+ *****************************************************************************/
+
+/** \class RooCBShape
+    \ingroup Roofit
+
+PDF implementing the Symmetrical Double-Sided Crystall Ball line shape.
+**/
+
+#include "RooSDSCBShape.h"
+#include "RooFit.h"
+
+#include "RooAbsReal.h"
+#include "RooRealVar.h"
+#include "RooMath.h"
+
+#include "TMath.h"
+
+#include "Riostream.h"
+#include "Riostream.h"
+#include <math.h>
+
+using namespace std;
+
+ClassImp(RooSDSCBShape);
+
+////////////////////////////////////////////////////////////////////////////////
+
+Double_t RooSDSCBShape::ApproxErf(Double_t arg) const
+{
+  static const double erflim = 5.0;
+  if( arg > erflim )
+    return 1.0;
+  if( arg < -erflim )
+    return -1.0;
+
+  return RooMath::erf(arg);
+}
+
+
+static const char rcsid[] =
+"$Id: RooSDSCBShape.C 1246 2013-04-02 02:51:06Z tskwarni $";
+
+////////////////////////////////////////////////////////////////////////////////
+
+RooSDSCBShape::RooSDSCBShape(const char *name, const char *title,
+		       RooAbsReal& _m, RooAbsReal& _m0, RooAbsReal& _sigma,
+		       RooAbsReal& _alpha, RooAbsReal& _n) :
+  RooAbsPdf(name, title),
+  m("m", "Dependent", this, _m),
+  m0("m0", "M0", this, _m0),
+  sigma("sigma", "Sigma", this, _sigma),
+  alpha("alpha", "Alpha", this, _alpha),
+  n("n", "Order", this, _n)
+{
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+RooSDSCBShape::RooSDSCBShape(const RooSDSCBShape& other, const char* name) :
+  RooAbsPdf(other, name), m("m", this, other.m), m0("m0", this, other.m0),
+  sigma("sigma", this, other.sigma), alpha("alpha", this, other.alpha),
+  n("n", this, other.n)
+{
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+Double_t RooSDSCBShape::evaluate() const {
+
+  Double_t t = fabs( (Double_t)( (m-m0)/sigma ) );
+  Double_t absAlpha = fabs( (Double_t)alpha);
+
+  if (t < absAlpha) {
+    return exp(-0.5*t*t);
+  }
+  else {
+    Double_t a =  TMath::Power(n/absAlpha,n)*exp(-0.5*absAlpha*absAlpha);
+    Double_t b= n/absAlpha - absAlpha;
+
+    return a/TMath::Power(b + t, n);
+  }
+
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+Int_t RooSDSCBShape::getAnalyticalIntegral(RooArgSet& allVars, RooArgSet& analVars, const char* /*rangeName*/) const
+{
+  if( matchArgs(allVars,analVars,m) )
+    return 1 ;
+
+  return 0;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+Double_t RooSDSCBShape::analyticalIntegral(Int_t code, const char* rangeName) const
+{
+  static const double sqrtPiOver2 = 1.2533141373;
+  static const double sqrt2 = 1.4142135624;
+
+  assert(code==1);
```

Comment:
Suppress the unused warning in release code:
```suggestion
  assert(code==1);
  (void)code;
```

Reply:
done. Also changed to `R__ASSERT`.